solidity/src/accounts/Account.sol

```diff
@@ -60,18 +63,26 @@ abstract contract Account is Ownable, IERC721Receiver {
             revert NotOwnerOrLibrary(msg.sender);
         }
 
+        // Ensure the target address is not zero
+        // This prevents accidental calls to the zero address which might lead to loss of funds or unexpected behavior
+        if (_target == address(0)) revert ZeroAddressTarget();
+
         (bool success, bytes memory returnData) = _target.call{value: _value}(_data);
 
         if (!success) {
-            assembly {
-                // returnData contains ABI-encoded error data:
-                // - First 32 bytes: Length of the error data (accessed via mload(returnData))
-                // - Next n bytes: Actual error data
+            if (returnData.length > 0) {
+                assembly {
+                    // returnData contains ABI-encoded error data:
+                    // - First 32 bytes: Length of the error data (accessed via mload(returnData))
+                    // - Next n bytes: Actual error data
```

Comment:
this is not related to the zero address execution right? feels like i've seen this before